hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/CassDefinitionsServiceImpl.java

```diff
@@ -782,81 +1047,155 @@ public Trigger copyTrigger(String tenantId, String triggerId, Map<String, String
-                newCondition = new StringCondition(newTrigger.getId(), c.getTriggerMode(),
-                        c.getConditionSetSize(), c.getConditionSetIndex(), dataIdMap.get(c.getDataId()),
-                        ((StringCondition)c).getOperator(), ((StringCondition)c).getPattern(),
-                        ((StringCondition)c).isIgnoreCase());
-            }
+            Condition newCondition = getMemberCondition(member, c, dataIdMap);
             if (newCondition != null) {
-                newCondition.setTenantId(newTrigger.getTenantId());
-                addCondition(newTrigger.getTenantId(), newTrigger.getId(), newCondition.getTriggerMode(), newCondition);
+                addCondition(newCondition);
             }
         }
 
         for (Dampening d : dampenings) {
-            Dampening newDampening = new Dampening(newTrigger.getId(), d.getTriggerMode(), d.getType(),
+            Dampening newDampening = new Dampening(member.getId(), d.getTriggerMode(), d.getType(),
                     d.getEvalTrueSetting(), d.getEvalTotalSetting(), d.getEvalTimeSetting());
-            newDampening.setTenantId(newTrigger.getTenantId());
+            newDampening.setTenantId(member.getTenantId());
+
+            addDampening(newDampening);
+        }
+
+        return member;
+    }
+
+    private Condition getMemberCondition(Trigger member, Condition groupCondition, Map<String, String> dataIdMap) {
+        Condition newCondition = null;
+        switch (groupCondition.getType()) {
+            case AVAILABILITY:
+                newCondition = new AvailabilityCondition(member.getId(), groupCondition.getTriggerMode(),
+                        groupCondition.getConditionSetSize(), groupCondition.getConditionSetIndex(),
+                        dataIdMap.get(groupCondition.getDataId()),
+                        ((AvailabilityCondition) groupCondition).getOperator());
+                break;
+            case COMPARE:
+                newCondition = new CompareCondition(member.getId(), groupCondition.getTriggerMode(),
+                        groupCondition.getConditionSetSize(), groupCondition.getConditionSetIndex(),
+                        dataIdMap.get(groupCondition.getDataId()),
+                        ((CompareCondition) groupCondition).getOperator(),
+                        ((CompareCondition) groupCondition).getData2Multiplier(),
+                        dataIdMap.get(((CompareCondition) groupCondition).getData2Id()));
+                break;
+            case EXTERNAL:
+                newCondition = new ExternalCondition(member.getId(), groupCondition.getTriggerMode(),
+                        groupCondition.getConditionSetSize(), groupCondition.getConditionSetIndex(),
+                        dataIdMap.get(groupCondition.getDataId()),
+                        ((ExternalCondition) groupCondition).getSystemId(),
+                        ((ExternalCondition) groupCondition).getExpression());
```

Comment:
Yes, good question. This is a tough one and there are a couple of ways we could go.

If you look at the expression used in the ExternalCondition in our hawkular-metrics integration (See Expression.java) you'll notice that we indeed have metricId in the expression:  "target:interval:func(metricId           op      threshold )[, period]".

But, like all conditions the ExternalCondition does have a dataId. And it's feasible that the dataId could be enough to allow the external alerter code handle things in its own code.  For example, consider a group external condition like this:

    External( dataId: $MetricIdToken$, expression: avg($MetricIdToken$ > 100.0),5 ).

When applied to Member1 it would look like:

    External( dataId: Member1MetricId, expression: avg($MetricIdToken$ > 100.0),5 ).

When processed by the external alerter it would be simple for the external alerter to replace $MetricIdToken$ in the expression with "Member1MetricId".  So, as long as the external alerter had support for groups, it could handle it.

On the other hand, it would also be pretty simple for us to do the same thing. We could apply the same token substitution and replace all occurrences of the same dataId token in the expression.

I'll actually add support for that now, it will probably be enough and will simplify the coding in the external alerter.